configure.ac

```
@@ -2713,39 +2713,34 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 dnl Check for special functions
 AC_FUNC_ALLOCA
 
-
-# TODO: remove this 'lib' prefix to match coding standard
-SQUID_AUTO_LIB(libcap,[Linux capabilities],[LIBCAP])
-AS_IF([test "x$with_libcap" != "xno"],[
+SQUID_AUTO_LIB(cap,[Linux capabilities],[LIBCAP])
+AS_IF([test "x$with_cap" != "xno"],[
+  SQUID_STATE_SAVE(squid_libcap_state)
   CXXFLAGS="$LIBCAP_CFLAGS $CXXFLAGS"
   LDFLAGS="$LIBCAP_PATH $LDFLAGS"
-  # cap_clear_flag is the most recent libcap function we require
-  AC_CHECK_HEADERS(sys/capability.h)
-  AC_CHECK_LIB(cap, cap_clear_flag)
+  PKG_CHECK_MODULES([LIBCAP],[libcap >= 2.09],[],[
+    # cap_clear_flag is the most recent libcap function we require
+    AC_CHECK_LIB([cap],[cap_clear_flag],[LIBCAP_LIBS="$LIBCAP_LIBS -lcap"])
+  ])
   SQUID_CHECK_FUNCTIONAL_LIBCAP2
-
-  # if it can't be supported..
-  AS_IF([test "x$ac_cv_header_sys_capability_h" = "xno" -o \
-     "x$ac_cv_lib_cap_cap_clear_flag" = "xno"],[
-    # and it was forced on: error
-    AS_IF([test "x$with_libcap" = "xyes"],[
-      AC_MSG_ERROR([libcap forced enabled but not available or not usable, requires libcap-2.09 or later])
-    ],[
-      # or it can't be supported: disable
-      with_libcap=no
-    ])
+  AC_MSG_NOTICE([libcap headers are ok: $squid_cv_sys_capability_works])
+  AS_IF([test "x$squid_cv_sys_capability_works" = "xno"],[LIBCAP_LIBS=""])
+  SQUID_STATE_ROLLBACK(squid_libcap_state)
+
+  AS_IF([test "x$LIBCAP_LIBS" != "x"],[
+    with_cap=yes
+    CXXFLAGS="$LIBCAP_CFLAGS $CXXFLAGS"
+    LDFLAGS="$LIBCAP_PATH $LDFLAGS"
```

This line looks like a copied typo to me; PKG_CHECK_MODULES() sets and we are then checking LIBCAP_LIBS rather than LIBCAP_PATH.

```suggestion
    LDFLAGS="$LIBCAP_LIBS $LDFLAGS"
```

There is a similar problem with a LIBCAP_PATH assignment in an earlier code blob (unchanged by this PR).

---

This particular AS_IF block (and equivalents for each --with-foo library option) is setting up the final/global build environment based on the test results. This line is informing the linker about the admins custom PATH location. Makefile.am use the *_LIBS only when needed to identify which bits to pull from there.

---

I now understand that we are adding LIBCAP_PATH because SQUID_AUTO_LIB sets it.

I still find it odd that there is a context with LIBCAP_PATH but without LIBCAP_LIBS; both variables may contain `-L` options. I understand that LIBCAP_LIBS may contain `-l` options, and that it may be wrong to add `-l` options to LDFLAGS, but those considerations do not make this look less odd. Perhaps we need a variable that contains just the `-L` options from `pkg-config`, or perhaps we should not update LDFLAGS at all, or perhaps there is a completely different solution. I did not study this problem enough to recommend a specific fix. I can only point out the inconsistency/oddity.

---

Most other libraries are pre-pending foo_PATH into foo_LIBS instead of global LDFLAGS. Testing that now.